functions/go/apply-setters/apply_setters.go

```diff
@@ -0,0 +1,241 @@
+}
+
+type Setter struct {
+	// Name is the name of the setter
+	Name string `json:"name,omitempty" yaml:"name,omitempty"`
+
+	// Value is the input value for setter
+	Value string `json:"value,omitempty" yaml:"value,omitempty"`
+}
+
+const SetterCommentIdentifier = "# kpt-set: "
+
+// Filter implements Set as a yaml.Filter
+func (as ApplySetters) Filter(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+	for i := range nodes {
+		err := accept(&as, nodes[i])
+		if err != nil {
+			return nil, errors.Wrap(err)
+		}
+	}
+	return nodes, nil
+}
+
+// visitMapping takes input mapping node, and performs following steps
+// checks if the key node of the input mapping node has line comment with SetterCommentIdentifier
+// checks if the value node is of sequence node type
+// if yes to both, resolves the setter value for the setter name in the line comment
+// replaces the existing sequence node with the new values provided by user
+func (as *ApplySetters) visitMapping(object *yaml.RNode) error {
+	return object.VisitFields(func(node *yaml.MapNode) error {
+		if node == nil || node.Key == nil || node.Key.YNode() == nil ||
+			node.Value == nil || node.Value.YNode() == nil {
+			return nil
+		}
+		// the aim of this method is to apply-setter for sequence nodes
+		if node.Value.YNode().Kind != yaml.SequenceNode {
+			// return if it is not a sequence node
+			return nil
+		}
+
+		setterPattern := extractSetterPattern(node.Key)
+		if setterPattern == "" {
+			// the node is not tagged with setter pattern
+			return nil
+		}
+
+		if !shouldSet(setterPattern, as.Setters) {
+			// this means there is no intent from user to modify this setter tagged resources
+			return nil
+		}
+
+		// since this setter pattern is found on sequence node, make sure that it is
+		// not interpolation of setters, it should be simple setter e.g. ${environments}
+		if !validArraySetterPattern(setterPattern) {
+			return errors.Errorf("setter interpolation is not allowed for arrays: %q", setterPattern)
+		}
+
+		// get the setter value for the setter name in the comment
+		sv := setterValue(as.Setters, setterPattern)
+
+		// parse the setter value as yaml node
+		rn, err := yaml.Parse(sv)
+		if err != nil {
+			return err
+		}
+
+		// the setter value must parse as sequence node
+		if rn.YNode().Kind != yaml.SequenceNode {
+			return errors.Errorf("input to array setter must be an array of values, but found %q", sv)
+		}
+
+		node.Value.YNode().Content = rn.YNode().Content
+		node.Value.YNode().Style = yaml.FoldedStyle
+		return nil
+	})
+}
+
+// visitScalar accepts the input scalar node and performs following steps,
+// checks if the line comment of input scalar node has prefix SetterCommentIdentifier
+// resolves the setter values for the setter name in the comment
+// replaces the existing value of the scalar node with the new value
+func (as *ApplySetters) visitScalar(object *yaml.RNode) error {
+	if object == nil || object.YNode() == nil {
```

Comment:
There are some helper methods in kyaml such as IsMissingOrNull, can you use one instead of rolling your own?